web/src/components/wizard/StepNavigation.tsx

```diff
@@ -13,20 +13,26 @@ const StepNavigation: React.FC<StepNavigationProps> = ({ currentStep }) => {
   const { prototypeSettings } = useConfig();
   const themeColor = prototypeSettings.themeColor;
 
+  // Navigation steps (validation is hidden but still part of the wizard flow)
   const steps = [
     { id: 'welcome', name: 'Welcome', icon: ClipboardList },
     { id: 'setup', name: 'Setup', icon: Settings },
-    { id: 'validation', name: 'Validation', icon: CheckCircle },
     { id: 'installation', name: mode === 'upgrade' ? 'Upgrade' : 'Installation', icon: Download },
     { id: 'completion', name: 'Completion', icon: CheckCircle },
   ];
 
+  // All wizard steps for progress calculation
+  const allSteps: WizardStep[] = ['welcome', 'setup', 'validation', 'installation', 'completion'];
+
   const getStepStatus = (step: { id: string }) => {
-    const stepIndex = steps.findIndex((s) => s.id === step.id);
-    const currentIndex = steps.findIndex((s) => s.id === currentStep);
+    const allStepIndex = allSteps.indexOf(step.id as WizardStep);
+    const currentAllStepIndex = allSteps.indexOf(currentStep);
```

Comment:
i think we can keep the old names, they're easier to grok
```suggestion
    const stepIndex = allSteps.indexOf(step.id as WizardStep);
    const currentStepIndex = allSteps.indexOf(currentStep);
```